java/org/apache/catalina/realm/DigestCredentialHandlerBase.java

```diff
@@ -288,7 +289,7 @@ protected String mutate(String inputCredentials, byte[] salt, int iterations, in
      * @return <code>true</code> if the strings are equal to each other, <code>false</code> otherwise.
      */
     public static boolean equals(final String s1, final String s2, final boolean ignoreCase) {
-        if (s1 == s2) {
+        if (Objects.equals(s1, s2)) {
```

Comment:
No. This is obviously a performance optimization and should not be replaced.

Reply:
memory adresses are compared here instead of string values